presto-main/src/main/java/com/facebook/presto/execution/SqlQueryExecution.java

```diff
@@ -422,17 +423,9 @@ private static Throwable failureCause(StageInfo stageInfo)
         @Override
         public SqlQueryExecution createQueryExecution(QueryId queryId, String query, Session session, Statement statement)
         {
-            int initialHashPartitions;
+            int initialHashPartitions = getHashPartitionCount(session, this.initialHashPartitions);
```

Comment:
Ya, I don't think it's particularly important. It could make sense to set big query and hash partitions, since big query effects other things too (like queue choice and join strategy)